packages/toolpad-components/src/Form.tsx

```diff
@@ -263,7 +260,12 @@ export function useFormInput<V>({
   React.useEffect(() => {
     if (
       form &&
-      !isEqual(validationProps, previousManualValidationPropsRef.current) &&
+      !(
+        validationProps.isInvalid === previousManualValidationPropsRef.current.isInvalid &&
```

Comment:
Sure, it's a bit more verbose though, but if we don't want to use any external utilities for deep equality it will still work.

Reply:
I'm adding a function for this